src/chunk_cache.rs

```diff
@@ -136,17 +167,10 @@ impl SimpleDiskCache {
         let name = name.to_string();
         let dir = PathBuf::from(dir);
         let path = dir.join(&name);
-        // Cache folder must either not exist or exist with metadata.
         if !dir.as_path().exists() {
             panic!("Cache parent dir {:?} must exist", dir)
         } else if path.exists() {
-            let file = path.join(SimpleDiskCache::STATE_FILE);
-            if !file.exists() {
-                panic!("Cache folder {:?} already exists without metadata", path.to_str())
-            }
-            // Wipe existing cache.
-            std::fs::remove_dir_all(&path).unwrap();
-            std::fs::create_dir(&path).unwrap();
+            panic!("Cache folder {:?} already exists", path.to_str())
```

Comment:
I don't think this needs to use the debug format when path is converted to a string
```suggestion
            panic!("Cache folder {} already exists", path.to_str())
```